pandas/core/internals/blocks.py

```diff
@@ -180,7 +184,19 @@ def _check_ndim(self, values, ndim):
         ValueError : the number of dimensions do not match
         """
         if ndim is None:
-            ndim = values.ndim
+            warnings.warn(
+                "Accepting ndim=None in the Block constructor is deprecated, "
```

Comment:
ahh i c, can you open an issue on arrow jira tracker and link to this PR for updating this